File: leetcode/python/9-palindrome-number.py
class Solution:
    def isPalindrome(self, x):
        
        outstring = str(abs(x))[::-1] #change to string and reverse it
        
        if int(outstring) > (2)**31: #check if |x| <= (2^31)-1
            outstring = '0'
        
        elif x < 0: # check if x negative or not
            #outstring = '-' + outstring
            return False

        y = int(outstring)

        if x == y:
            return True
        else:
            return False

Solution.isPalindrome(Solution, -121)
        

File: leetcode/go/9-palindrome-number.go
func isPalindrome(x int) bool {
    if x < 0 {
		return false
	}

	rev := 0
	ori := x

	for x > 0 {
		rev = rev * 10 + x % 10
		x = x / 10
	}

	return rev == ori
}